### PR TITLE
CSUB-145 Skim events

### PR DESCRIPTION
**Pull request by CAGS295**
cross-ref gluwa/creditcoin#164

### CODE REVIEW

File: src/extrinsics/common.ts [hunk 1]
```diff
@@ -51,24 +51,29 @@ export const handleTransaction = (
...
 
-    const [id, dataItem] = sourceEvents.event.data;
-    const itemId = id.toJSON() as TupleId | string;
-    const sourceItem = api.createType(creditcoinType, dataItem) as SourceType;
-    return { itemId, item: transform(sourceItem) };
+    const [id, codecItem] = sourceEvents.event.data;
+
+    const itemId = id.toJSON() as unknown as IdType;
+
+    const transformWrapper = (dataItem: Codec, transform_fn: (data: SourceType) => ItemType) => {
```
Comment (0xjeremyfrank) [inline src/extrinsics/common.ts]:
```suggestion
    const transformWrapper = (dataItem: Codec, transformFn: (data: SourceType) => ItemType) => {
```

or something like that. Gotta use camelCase with javaScript.

File: src/extrinsics/common.ts [hunk 2]
```diff
@@ -51,24 +51,29 @@ export const handleTransaction = (
...
+        const sourceItem = api.createType(creditcoinType, dataItem) as SourceType;
+        return transform_fn(sourceItem);
+    };
+
+    if (codecItem != null) {
```
Comment (0xjeremyfrank) [inline src/extrinsics/common.ts]:
```suggestion
    if (codecItem) {
```
or you could use a ternary:
```
return codecItem ? { itemId, item:transformWrapper(codecItem, transform)} : {itemId}
```

File: src/model.ts [hunk 3]
```diff
@@ -82,24 +82,31 @@ export type DealOrder = {
...
-export type DealOrderAdded = DealOrderEvent;
-export type DealOrderLocked = DealOrderEvent;
+type EventReturnDataType<DataType> = {
+    item: DataType;
+};
 
-export type TransferId = string;
+export type EventReturnJoinType<Id, Data> = EventReturnIdType<Id> & EventReturnDataType<Data>;
+
+export type EventReturnType<I, D> = EventReturnIdType<I> | EventReturnJoinType<I, D>;
```
Comment (0xjeremyfrank) [inline src/model.ts]:
```suggestion
export type EventReturnType<Id, Data> = EventReturnIdType<Id> | EventReturnJoinType<Id, Data>;
```
for consistency

File: src/model.ts [hunk 4]
```diff
@@ -82,24 +82,31 @@ export type DealOrder = {
...
-    dealOrder: DealOrder;
+type EventReturnIdType<IdType> = {
+    itemId: IdType;
 };
 
-export type DealOrderClosed = DealOrderEvent;
-export type DealOrderFunded = DealOrderEvent;
-export type DealOrderAdded = DealOrderEvent;
-export type DealOrderLocked = DealOrderEvent;
+type EventReturnDataType<DataType> = {
```
Comment (0xjeremyfrank) [inline src/model.ts]: Change generic to `Data` for consistency.

**Review by 0xjeremyfrank — CHANGES_REQUESTED** [review]
Cosmetic nitpicks. I like the approach!

**Review by 0xjeremyfrank — APPROVED** [review]
looks good

**Review by atodorov — APPROVED** [review]

**Pull request merged**